records_mover/db/driver.py

```diff
@@ -42,43 +65,63 @@ def schema_sql(self,
         """
         # http://docs.sqlalchemy.org/en/latest/core/reflection.html
         table_obj = self.table(schema, table)
-        return str(CreateTable(table_obj, bind=self.db))
+        return str(CreateTable(table_obj))
 
     def varchar_length_is_in_chars(self) -> bool:
         """True if the 'n' in VARCHAR(n) is represented in natural language
         characters, rather than in post-encoding bytes.  This varies by database -
         override it to control"""
         return False
 
-    def set_grant_permissions_for_groups(self, schema_name: str, table: str,
+    def set_grant_permissions_for_groups(self,
+                                         schema_name: str,
+                                         table: str,
                                          groups: Dict[str, List[str]],
-                                         db: Union[sqlalchemy.engine.Engine,
-                                                   sqlalchemy.engine.Connection]) -> None:
-        schema_and_table: str = quote_schema_and_table(self.db.engine, schema_name, table)
+                                         db: Optional[Union[sqlalchemy.engine.Engine,
+                                                      sqlalchemy.engine.Connection]],
+                                         db_conn: Optional[sqlalchemy.engine.Connection] = None,
+                                         db_engine: Optional[sqlalchemy.engine.Engine] = None
+                                         ) -> None:
+        db, db_conn, db_engine = check_db_conn_engine(db=db, db_conn=db_conn, db_engine=db_engine)
+        schema_and_table: str = quote_schema_and_table(None, schema_name, table,
+                                                       db_engine=self.db_engine)
         for perm_type in groups:
             groups_list = groups[perm_type]
             for group in groups_list:
-                group_name: str = quote_group_name(self.db.engine, group)
+                group_name: str = quote_group_name(None, group, db_engine=self.db_engine)
                 if not perm_type.isalpha():
                     raise TypeError("Please make sure your permission types"
                                     " are an acceptable value.")
                 perms_sql = f'GRANT {perm_type} ON TABLE {schema_and_table} TO {group_name}'
-                db.execute(perms_sql)
+                if db_conn:
+                    db_conn.execute(perms_sql)
+                else:
+                    with db_engine.connect() as conn:
+                        conn.execute(perms_sql)
```

Comment:
would be nice to be able to leverage `self.get_db_conn()` for this, but i see that the db we have here isn't `self` so i  don't think we can. eugh

Reply:
yeah, was thinking the same thing but agree that we're just going to have to go with whatever db is passed